src/components/views/location/LocationButton.tsx

```diff
@@ -75,6 +82,38 @@ export const LocationButton: React.FC<IProps> = (
     </React.Fragment>;
 };
 
+const shareLocation = (client: MatrixClient, roomId: string, openMenu: () => void) =>
+    (uri: string, ts: number) => {
+        if (!uri) return false;
+        try {
+            const text = textForLocation(uri, ts, null);
+            client.sendMessage(
+                roomId,
+                makeLocationContent(text, uri, ts, null),
+            );
+        } catch (e) {
+            logger.error("We couldn’t send your location", e);
+            Modal.createTrackedDialog(
+                'We couldn’t send your location',
+                '',
+                QuestionDialog,
+                {
+                    title: _t("We couldn’t send your location"),
+                    description: _t(
+                        "Element could not send your location. Please try again later."),
+                    button: _t('Try again'),
+                    cancelButton: _t('Cancel'),
+                    onFinished: (tryAgain: boolean) => {
+                        if (tryAgain) {
+                            openMenu();
+                        }
+                    },
+                },
+            );
+        }
+        return true;
+    };
```

Comment:
```suggestion
const shareLocation = (
    client: MatrixClient,
    roomId: string,
    openMenu: () => void,
) => (
    uri: string,
    ts: number,
): boolean => {
    if (!uri) return false;
    try {
        const text = textForLocation(uri, ts, null);
        client.sendMessage(
            roomId,
            makeLocationContent(text, uri, ts, null),
        );
    } catch (e) {
        logger.error("We couldn’t send your location", e);
        Modal.createTrackedDialog(
            'We couldn’t send your location',
            '',
            QuestionDialog,
            {
                title: _t("We couldn’t send your location"),
                description: _t(
                    "Element could not send your location. Please try again later."),
                button: _t('Try again'),
                cancelButton: _t('Cancel'),
                onFinished: (tryAgain: boolean) => {
                    if (tryAgain) {
                        openMenu();
                    }
                },
            },
        );
    }
    return true;
};
```

perhaps something like this, I don't think its massively more readable so maybe not using the shorthand inline return on the arrow func here and using braces and an actual return might be cleaner

Reply:
I attempted to make it better by breaking the args into separate variables in bdcf002 .